app/services/parse_csv_service.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require 'csv'
+require 'logger'
+
+module Services
+  class ParseCsvService
+    def initialize(csv_data)
+      @csv_data = csv_data
+    end
+
+    def call
+      rows = CSV.parse(@csv_data, headers: false)
+      rows.map do |row|
+        validate_row(row)
+      end.compact
+    rescue CSV::MalformedCSVError => e
+      raise "Error parsing CSV: #{e.message}"
```

Comment:
same as with `FetchCsv` service

Reply:
It should throw specific errors with a generic message.